src/ua.c

```diff
@@ -835,9 +835,10 @@ void ua_hangup(struct ua *ua, struct call *call,
 			return;
 	}
 
-	(void)call_hangup(call, scode, reason);
+	ua_event(ua, UA_EVENT_CALL_CLOSED, call,
+	         reason ? reason : "Connection reset by user");
 
-	ua_event(ua, UA_EVENT_CALL_CLOSED, call, reason);
+	(void)call_hangup(call, scode, reason);
```

Comment:
is this change needed, what is the reason for it ?

Reply:
aef93f0a2535948606b04ffd311c863f6da4422d

````
    ua: emit 'call_hangup' event before calling the function

    otherwise the call session is dereferenced and call_id inaccesible.
````

`call_hangup()` releases `call->sess`, which is required for `call_id()` on the later `UA_EVENT_CALL_CLOSED` event generation.

https://github.com/jmillan/baresip/blob/ctrl_tcp/src/call.c#L759

Reply:
I fixed it in a different way, by saving the callid string. Here is the commit

9d0b0fec87691312ade86b4d330265b5ca59c997

can you please revert this change ?